src/server/server_family.cc

```diff
@@ -114,12 +118,73 @@ void SubstituteFilenameTsPlaceholder(fs::path* filename, std::string_view replac
   *filename = absl::StrReplaceAll(filename->string(), {{"{timestamp}", replacement}});
 }
 
-string InferLoadFile(fs::path data_dir) {
+bool IsCloudPath(string_view path) {
+  return absl::StartsWith(path, "s3://");
+}
+
+// Returns bucket_name, obj_path for an s3 path.
+optional<pair<string_view, string_view>> GetBucketPath(string_view path) {
```

Comment:
Given this is an internal function (side note: maybe wrap it in an anonymous `namespace`?), I'd define a struct for its return type, i.e. `struct AwsData { string_view bucket_name; string_view obj_path; };`

BTW consider using `ABSL_ATTRIBUTE_LIFETIME_BOUND` (see [here](https://github.com/abseil/abseil-cpp/blob/master/absl/base/attributes.h#L732)) to indicate that the returned value cannot surpass the input's lifetime

Reply:
it works but I did not like how much visual clutter it adds. Decided to just return strings.